### PR TITLE
Feature/domain resolution

### PR DESCRIPTION
**Pull request by JohnnyJumper**
## Integration with Unstoppabledomains
Using @unstoppabledomains/resolution to do the resolution for blockchain domains.
Screenshots with the results bellow

photo5028715811338037344
photo5028715811338037343
photo5028715811338037342

## Changes
- package.json
- src/components/modals/AddressModal.js
- src/locales/en_US.js
- src/locales/strings/enUS.json
- src/util/utils.js

## Enviroment tested at
 - Android phone ( OnePlus T6 )
 - Android version: 9
 - OxygenOS version 9.0.17

### CODE REVIEW

File: src/components/modals/AddressModal.js [hunk 1]
```diff
@@ -60,6 +65,65 @@ export class AddressModal extends Component<AddressModalProps, AddressModalState
...
 
+  setStatusLabel = (status: string) => {
+    this.setState({ statusLabel: status })
+  }
+
+  setCryptoAddress = (address: string) => {
+    this.setState({ cryptoAddress: address })
+  }
+
+  getServiceName = (resolutionObject: Resolution, domain: string) => {
```
Comment (paullinator) [inline src/components/modals/AddressModal.js]: unused?

File: src/components/modals/AddressModal.js [hunk 2]
```diff
@@ -11,16 +11,20 @@ import * as Constants from '../../constants/indexConstants'
...
 import styles from '../../styles/scenes/ScaneStyle'
 import { colors as COLORS } from '../../theme/variables/airbitz.js'
+import ResolutionError, { ResolutionErrorCode} from '../common/ResolutionError.js'
 
 // INTERACTIVE_MODAL /////////////////////////////////////////////////////////////////////////////
 type AddressModalProps = {
   onDone: any => void,
-  coreWallet: EdgeCurrencyWallet
+  coreWallet: EdgeCurrencyWallet,
+  currencyCode: string
```
Comment (paullinator) [inline src/components/modals/AddressModal.js]: unused?

File: src/components/modals/AddressModal.js [hunk 3]
```diff
@@ -11,16 +11,20 @@ import * as Constants from '../../constants/indexConstants'
...
 }
 
 type AddressModalState = {
   clipboard: string,
-  uri: string
+  uri: string,
+  statusLabel: string,
+  cryptoAddress?: string
```
Comment (paullinator) [inline src/components/modals/AddressModal.js]: unused

File: src/components/modals/AddressModal.js [hunk 4]
```diff
@@ -60,6 +65,65 @@ export class AddressModal extends Component<AddressModalProps, AddressModalState
     }
   }
 
+  setStatusLabel = (status: string) => {
+    this.setState({ statusLabel: status })
+  }
+
+  setCryptoAddress = (address: string) => {
```
Comment (paullinator) [inline src/components/modals/AddressModal.js]: unused

File: src/components/modals/AddressModal.js (hunk 2 above)
Comment (JohnnyJumper) [reply src/components/modals/AddressModal.js]: This is needed for the error code in cases when some domain doesn't have a certain ticker configured. 

Example of the error message would be: 
" Domain brad.crypto has no BTC address attached to it "

File: src/components/modals/AddressModal.js [hunk 5]
```diff
@@ -89,11 +148,11 @@ export class AddressModal extends Component<AddressModalProps, AddressModalState
...
                 style={MaterialInputStyle}
                 value={uri}
-                onChangeText={this.updateUri}
+                onChangeText={this.onChangeTextDelayed}
                 error={''}
                 placeholder={s.strings.fragment_send_send_to_hint}
-                label={s.strings.fragment_send_send_to_hint}
-                onSubmit={() => this.props.onDone(uri)}
+                label={statusLabel}
+                onSubmit={() => this.props.onDone(cryptoAddress)}
```
Comment (paullinator) [inline src/components/modals/AddressModal.js]: It looks like we will never send the URI to the onDone() Call back which is necessary if domain resolution does not succeed

**Pull request merged**